kubescape/helm-charts
# Fix that inspektor-gadget couldn't call api/v1/nodes/NODE_NAME/proxy/configz

The nodes/proxy resource was missing from the `node-agent` RBAC `ClusterRole`.

## Overview

Adds the `nodes/proxy` resource to the list of resources on the `node-agent ClusterRole`.

---

## Additional info

This PR: https://github.com/inspektor-gadget/inspektor-gadget/pull/2794/ in the `Inspektor-gadget` project ( that `KubeScape` seems to use for generating `network policies` ) - determines the container runtime socket path by querying the kubelet API at the node level. Therefore the need for the permission on `nodes/proxy`.

As far as I've been able to conclude signaling/setting the `INSPEKTOR_GADGET_CONTAINERD_SOCKETPATH` no longer _does_ anything. At least when configuring that on the `node-agent` `DaemonSet deployment` it isn't picked up as far as I can see.

I had a good long forth and back on Slack on this matter. See [this link](https://cloud-native.slack.com/archives/C04EY3ZF9GE/p1730923553266949).

---

## How to Test

- if one does **NOT** have this permission the following error will reveal its _ugly_ face:

```text
{"level":"info","ts":"2024-11-06T19:42:00Z","msg":"credentials loaded","accountLength":0}                                                 W1106 19:42:00.795912 3943570 client_config.go:659] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.                                                                                                                              {"level":"info","ts":"2024-11-06T19:42:00Z","msg":"Detected container runtime","containerRuntime":"containerd"}                           {"level":"info","ts":"2024-11-06T19:42:00Z","msg":"starting health manager","port":7888}                                                  time="2024-11-06T19:42:00Z" level=warning msg="Runtime enricher (containerd): couldn't get current containers: listing containers with request &ListContainersRequest{Filter:nil,}: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial unix /run/containerd/containerd.sock: connect: no such file or directory\""                                                          time="2024-11-06T19:42:01Z" level=warning msg="Failed to retrieve socket path for runtime client from kubelet: getting /configz: fetching /configz from \"test-test-worker-106\": unknown (get nodes test-test-worker-106). Falling back to default container runtime"              {"level":"info","ts":"2024-11-06T19:42:03Z","msg":"main container handler started"}
```

_Specifically_:

```text
time="2024-11-06T19:42:00Z" level=warning msg="Runtime enricher (containerd): couldn't get current containers: listing containers with request &ListContainersRequest{Filter:nil,}: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial unix /run/containerd/containerd.sock: connect: no such file or directory\""                                                          time="2024-11-06T19:42:01Z" level=warning msg="Failed to retrieve socket path for runtime client from kubelet: getting /configz: fetching /configz from \"test-test-worker-106\": unknown (get nodes test-test-worker-106). Falling back to default container runtime"              
```

Is interesting. Where the `unknown` error is the tattle teller.

*When working the output looks like this in X `node-agent Pod`.*

```text
{"level":"info","ts":"2024-11-08T13:46:02Z","msg":"Detected container runtime","containerRuntime":"containerd"}
{"level":"info","ts":"2024-11-08T13:46:02Z","msg":"starting health manager","port":7888}
time="2024-11-08T13:46:02Z" level=warning msg="Runtime enricher (containerd): couldn't get current containers: listing containers with request &ListContainersRequest{Filter:nil,}: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial unix /run/containerd/containerd.sock: connect: no such file or directory\""
time="2024-11-08T13:46:03Z" level=info msg="using the detected container runtime socket path from Kubelet's config: /run/k3s/containerd/containerd.sock"
{"level":"info","ts":"2024-11-08T13:46:05Z","msg":"main container handler started"}
{"level":"debug","ts":"2024-11-08T13:46:05Z","msg":"WatchHandler - getting existing objects from storage","resource":"seccompprofiles"}
{"level":"debug","ts":"2024-11-08T13:46:05Z","msg":"WatchHandler - starting watch","resource":"pods"}
{"level":"debug","ts":"2024-11-08T13:46:05Z","msg":"WatchHandler - starting watch","resource":"seccompprofiles"}

Now we see:

```text
time="2024-11-08T13:46:03Z" level=info msg="using the detected container runtime socket path from Kubelet's config: /run/k3s/containerd/containerd.sock"
```

Which is very nice :+1: 

---

## Checklist before requesting a review

- [NOT_RELEVANT] My code follows the style guidelines of this project
- [NOT_RELEVANT] I have commented on my code, particularly in hard-to-understand areas
- [NOT_RELEVANT] If it is a core feature, I have added thorough tests.
- [NOT_RELEVANT] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)** < I can't see any dev branch. Please guide me here if there's another branch to create the PR on. Checked other `PR's` and they seem to use the `main branch`.
